screens/ProfileScreen/ProfileScreen.tsx

```diff
@@ -0,0 +1,183 @@
+import { Caption, Headline, Title } from "@components/Text"
+import { OutlinedButton, PrimaryButton } from "@components/common/Buttons"
+import ExpandableText from "@components/common/ExpandableText"
+import { ToastWithIcon } from "@components/common/Toasts"
+import ProfileImage from "@components/profileImageComponents/ProfileImage"
+import { User } from "@lib/User"
+import { useState } from "react"
+import { FlatList, ListRenderItemInfo, ScrollView, StyleSheet, TouchableOpacity, View } from "react-native"
+import EventPager from "./EventCarousel"
+import AllEventsModal from "./AllEventsModal"
+import { AppStyles } from "@lib/AppColorStyle"
+import { EventCard } from "@components/eventCard/EventCard"
+import { CurrentUserEvent } from "@lib/events"
+
+export interface ProfileScreenProps {
+  user: User
+}
+
+const NUM_EVENTS_SHOWN = 20
+
+const ProfileScreen = ({ user }: ProfileScreenProps) => {
+  const [userStatus, setUserStatus] = useState(user.userStatus)
+  const [modalVisible, setModalVisible] = useState(false)
+  const [requestSent, setRequestSent] = useState(
+    !!(userStatus === "friend-request-pending" || userStatus === "friends")
+  )
+  const colorText = "#4285F4"
+  const firstEvents = user.events.slice(0, NUM_EVENTS_SHOWN)
+
+  const setStatus = () => {
+    setUserStatus("friend-request-pending")
+  }
+
+  const openModal = () => {
+    setModalVisible(true)
+  }
+
+  return (
+    <View style={styles.container}>
+      <AllEventsModal
+        username={user.username}
+        visible={modalVisible}
+        setVisible={setModalVisible}
+        events={user.events}
```

Comment:
We'll probably have /user/events as a separate query within the events modal instead of adding it here, in case the list is too long @mhayes853